### DISCUSSION TITLE
Do `pin_count` and `dirty` have to be in the `Page` interface in a DB?

### DISCUSSION THREAD

**Discussion by GavinRay97** [cmu-db/bustub#290] [Q&A] [answered]
I'm trying to understand whether there is a special reason for putting `pin_count` and `dirty` on the `Page` interface -- or if it was just a convenient design choice?

Why I ask this question is because, if I understand right, this information is only relevant as part of the `"frame table"` in the Buffer Manager?

```cpp
Page my_page = Page {};
// Here, the pin_count and dirty bit doesn't mean anything
```

Is the below (or something similar) also a valid design?

```java
abstract class Page {
    byte[] data;
    PageId pageId;
    int lsn;
}

class Frame {
   PageId pageId;
   FrameId frameId;
   int pinCount;
   boolean isDirty;
}

class BufferPoolManager {
   Map<PageId, Frame> frameTable;
}
```

**Comment by skyzh** [comment, marked as answer]
The design looks good to me! That's all about design choices.

**Reply by GavinRay97** [reply]
Thanks for the feedback, just wanted to make sure I understand 👍
